### PR TITLE
Ui/ttl form

### PR DESCRIPTION
**Pull request by chelshaw**
This component will be used in upcoming work where a Time To Live (TTL) value is a required input. It has the same underlying functionality as the TtlPicker2 (minus the toggle) so I chose to move the base functionality for the unit and time into this this simpler TTL version, and extend the TtlPicker2 from that. 

ttl-form

### CODE REVIEW

File: ui/lib/core/addon/components/ttl-form.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
+/**
+ * @module TtlForm
+ * TtlForm components are used to enter a Time To Live (TTL) input.
+ * This component does not include a label and is designed to take
+ * a time and unit, and pass an object including seconds and
+ * timestring when those two values are changed.
+ *
+ * @example
+ * ```js
+ * <TtlForm @onChange={action handleChange} @unit={{m}}/>
```
Comment (Monkeychip) [inline ui/lib/core/addon/components/ttl-form.js:10]: curious if the double curly brackets around m should be one?

File: ui/lib/core/addon/components/ttl-form.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    this.setProperties({
+      time: newTime,
+      unit: newUnit,
+    });
+  },
+  updateTime: task(function*(newTime) {
+    this.set('errorMessage', '');
+    let parsedTime;
+    parsedTime = parseInt(newTime, 10);
+    if (!newTime) {
```
Comment (Monkeychip) [inline ui/lib/core/addon/components/ttl-form.js:73]: nice errorMessage handling!

File: ui/lib/core/addon/components/ttl-form.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      this.set('errorMessage', 'This field is required');
+      return;
+    } else if (Number.isNaN(parsedTime)) {
+      this.set('errorMessage', 'Value must be a number');
+      return;
+    }
+    this.set('time', parsedTime);
+    this.handleChange();
+    if (Ember.testing) {
```
Comment (Monkeychip) [inline ui/lib/core/addon/components/ttl-form.js:82]: oh the fun bits of concurrency and testing; I love concurrency, I hate testing it.

File: ui/lib/core/addon/components/ttl-form.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+      { label: 'days', value: 'd' },
+    ];
+  }),
+  handleChange() {
+    let { time, unit, seconds } = this.getProperties('time', 'unit', 'seconds');
+    const ttl = {
+      seconds,
+      timeString: time + unit,
+    };
+    this.onChange(ttl);
```
Comment (noelledaley) [inline ui/lib/core/addon/components/ttl-form.js:60]: do you need a `return` here?

File: ui/lib/core/stories/ttl-form.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
+<!--THIS FILE IS AUTO GENERATED. This file is generated from JSDoc comments in lib/core/addon/components/ttl-form.js. To make changes, first edit that file and run "yarn gen-story-md ttl-form" to re-generate the content.-->
```
Comment (noelledaley) [inline ui/lib/core/stories/ttl-form.md:1]: pro-tip: now that we've upgraded storybook you should be able to take advantage of Storybook Docs, which will generate these pages for you 👯‍♀️

https://github.com/storybookjs/storybook/tree/master/addons/docs

**Comment by noelledaley** [comment]
i really like what you've done here by extending the `TTLForm` as a base class. out of curiosity, could you do the same thing with the old `TTLPicker`?

File: ui/lib/core/addon/components/ttl-form.js (hunk 1 above)
Comment (chelshaw) [reply ui/lib/core/addon/components/ttl-form.js:10]: Should actually just be quotes, good catch

File: ui/lib/core/addon/components/ttl-form.js (hunk 4 above)
Comment (chelshaw) [reply ui/lib/core/addon/components/ttl-form.js:60]: I don't think so -- it's not an async function and nothing is waiting on its value. It's essentially just calculating and triggering a side-effect.

**Comment by chelshaw** [comment]
> i really like what you've done here by extending the `TTLForm` as a base class. out of curiosity, could you do the same thing with the old `TTLPicker`?

Good question! I decided against this because `TtlForm` and `TtlPicker2` diverge from the original `TtlPicker` in terms of expected inputs and outputs. It is my full intention to replace all of the TtlPicker instances with one of the updates ones, and then remove that component (and rename TtlPicker2)

**Pull request merged**